scala-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/scala/ScalaInterpreter.scala

```diff
@@ -328,6 +466,10 @@ class ScalaInterpreter(private val config:Config = ConfigFactory.load) extends I
 
 object ScalaInterpreter {
 
+  val NamedResult = """(\w+):\s+([^=]+)\s+=\s*(.*)""".r
```

Comment:
No, this won't work with function types since it assumes that types are any series of characters before the =. In the short term, we may want to have a simpler match that just grabs the variable name. Since we confirm that it is a variable by looking it up in the interpreter anyway, it is probably easier to get the variable and then inspect it to get the information that could be parsed here.

Long term, I don't think that parsing a stream is the way to go but I don't see a much better way with the current scala REPL API.